src/vaadin-text-field-mixin.html

```diff
@@ -792,6 +825,8 @@
 
     clear() {
       this.value = '';
+
+      this._inputElementValue = '';
```

Comment:
It shouldn't be necessary for text fields because there is no situation when `this.value` would hold an empty string and `this._inputElementValue` would not. This line might make sense for time-picker or date-picker.